src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,240 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Class that deals with making sense of the user command
+ */
+public class Parser {
+    /**
+     * Parses the input and returns if the scanner should scan a new line
+     *
+     * @param first The users input
+     * @param lst The tasklist object
+     * @param store The storage object
+     * @return boolean A boolean that represents whether the scanner should continue with
+     *                 a new line
+     */
+    public static boolean parse(String first, TaskList lst, Storage store) {
+        try {
+            if (first.equals("bye")) {
+                Ui.byeMessage();
+                return true;
+            } else if (first.equals("list")) {
+                listMessage(lst);
+                return false;
+            } else if (first.length() == 6 && first.substring(0, 4).equals("mark")) {
+                markMessage(first, lst, store);
+                return false;
+            } else if (first.length() == 8 && first.substring(0, 6).equals("unmark")) {
+                unmarkMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 4 && first.substring(0, 4).equals("todo")) {
+                todoMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 8 && first.substring(0, 8).equals("deadline")) {
+                deadlineMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 5 && first.substring(0, 5).equals("event")) {
+                eventMessage(first, lst, store);
+                return false;
+            } else if (first.length() == 8 && first.substring(0, 6).equals("delete")) {
+                deleteMessage(first, lst, store);
+                return false;
+            } else if (first.length() >= 4 && first.substring(0, 4).equals("find")) {
+                findMessage(first, lst, store);
+                return false;
+            }
```

Comment:
LGTM, can consider using switch case to reduce the number of lines

Reply:
Thanks for the suggestions will implement that.